simulator/model.go

```diff
@@ -530,8 +530,11 @@ func (m *Model) CreateInfrastructure(ctx *Context) error {
 
 	// addMachine returns a func to create a VM.
 	addMachine := func(prefix string, host *object.HostSystem, pool *object.ResourcePool, folders *object.DatacenterFolders) {
-		nic := esx.EthernetCard
-		nic.Backing = vmnet
+		nic := func() types.BaseVirtualDevice {
+			nic := esx.EthernetCard // shallow copy for each vm
```

Comment:
Is a top level shallow copy sufficient or do we need to go a level deeper?
We're replacing the Backing, but `Key` and `Connectable.Connected` look like they'll remain shared.
Does vcsim support multiple NICs per VM? If so I guess we're not checking that Key is unique.